mcstatus/motd.py

```diff
@@ -0,0 +1,549 @@
+        """Parse hex or RGB color, from hex or RGB (one of them can be ``None``).
+
+        Ensures that ``WebColor.hex`` starts with '#', and generate RGB color for it,
+        if ``WebColor.rgb`` is None (default value). Or generate hex value from RGB, if
+        ``WebColor.hex`` is None (also default value).
+
+        :raises ValueError: When hex color isn't correct, so we can't generate RGB for it.
+        :raises TypeError: When hex and RGB colors is ``None``.
+        :returns: New ``WebColor`` instance.
+        """
+        if hex is None and rgb is None:
+            raise TypeError("`hex` and `rgb` parameters is None, we need one of them.")
+
+        if hex is None and rgb is not None:
+            if not all(256 > e > -1 for e in rgb):
+                raise ValueError("One of RGB colors is less than zero or more than 255: " + str(rgb))
+            hex = "#%02x%02x%02x" % rgb
+        elif hex is not None and rgb is None:
+            if not hex.startswith("#"):
+                hex = "#" + hex
+            try:
+                rgb = tuple(int(hex.lstrip("#")[i : i + 2], 16) for i in (0, 2, 4))
+            except ValueError:
+                raise ValueError(f"Incorrect hex color '{hex}', failed to generate RGB.")
+
+        return cls(hex=hex, rgb=rgb)  # type: ignore[assignment]
+
+
+@dataclass
+class TranslateString:
+    """Represents a ``translate`` field in server's answer.
+
+    This just exist, and completely ignored by our end-parsers (``ansi``, ``html`` etc. properties).
+    You can parse them with ``Motd.parsed`` attribute.
+
+    .. seealso:: `Minecraft's wiki. <https://minecraft.fandom.com/wiki/Raw_JSON_text_format#Translated_Text>`_
+    """
+
+    id: str
+
+
+ParsedMotdItem: TypeAlias = Union[Formatting, MinecraftColor, WebColor, TranslateString, str]
+
+
+@dataclass
+class Motd:
+    """Represents parsed MOTD.
+
+    Basing on this class, you can write your own MOTD-to-something parser, see ``parsed`` field.
+    """
+
+    parsed: list[ParsedMotdItem]
+    """Parsed MOTD, which then will be transformed.
+
+    Bases on this field, you can easily write your own MOTD-to-something parser.
+    """
+    bedrock: bool = False
+    """Is server Bedrock Edition? Some details may change in work of this class."""
+
+    @classmethod
+    def parse(cls, raw: Union[dict, list, str], *, bedrock: bool = False) -> Self:
+        """Parse a raw MOTD to less raw MOTD (``Motd.parsed`` field).
+
+        :param raw: Raw MOTD, directly from server.
+        :param bedrock: Is server Bedrock Edition? Nothing changes here, just sets attribute.
+        :returns: New ``MOTD`` instance.
+        """
+        if isinstance(raw, str):
+            return cls(cls._parse_as_str(raw, bedrock=bedrock), bedrock)
+
+        if isinstance(raw, list):
+            raw = {"extra": raw}
+
+        return cls(cls._parse_as_dict(raw), bedrock)
```

Comment:
A slightly less repetitive version

```suggestion
        if isinstance(raw, list):
            raw = {"extra": raw}
           
        if isinstance(raw, str):
            parsed = cls._parse_as_str(raw, bedrock=bedrock)
        else:
            parsed = cls._parse_as_dict(raw)

        return cls(parsed, bedrock)
```

Reply:
Fixed in [last force push](https://github.com/py-mine/mcstatus/compare/8b2935c714ffd2ac802fe9538a10dd356d7ae991..aaa48785fdf6a1cfc88b93343d12c3008aced2cc).